content/examples/databricks-cluster-local.Rmd

```diff
@@ -0,0 +1,160 @@
+on the Spark driver node within your Databricks cluster.
+
+The recommended method for installing RStudio Server Pro to the Spark driver
+node is via SSH. However, an alternative method is available if you are not able
+to access the Spark driver node via SSH.
+
+## Configure SSH access to the Spark driver node
+
+Configure SSH access to the Spark driver node in Databricks by following the
+steps in the [SSH access to clusters section of the Databricks Cluster
+configurations
+documentation](https://docs.databricks.com/clusters/configure.html#ssh-access-to-clusters).
+
+**Note:** If you are unable to configure SSH access or connect to the Spark
+driver node via SSH, then you can follow the steps in the [Get started with
+RStudio Server Pro section of the RStudio on Databricks
+documentation](https://docs.databricks.com/spark/latest/sparkr/rstudio.html#get-started-with-rstudio-server-pro)
+to install RStudio Server Pro from a Databricks notebook, then skip to the
+section on "Using sparklyr".
+
+## Connect to the Spark driver node via SSH
+
+Connect to the Spark driver node via SSH on port 2200 by using the following
+command on your local machine:
+
+```
+ssh ubuntu@<spark-driver-node-address> -p 2200 -i <path-to-private-SSH-key>
+```
+
+Replace `<spark-driver-node-address>` with the DNS name or IP address of the
+Spark driver node, and `<path-to-private-SSH-key>` with the path to your private
+SSH key on your local machine.
+
+## Install RStudio Server Pro on the Spark driver node
+
+After you SSH into the Spark driver node, then you can follow the steps to
+[install RStudio Server Pro in the RStudio documentation](https://docs.rstudio.com/rsp/installation/).
+In the installation steps, you can select Ubuntu as the target Linux
+distribution.
+
+## Configure RStudio Server Pro
+
+The following configuration steps are required to be able to use RStudio Server
+Pro with Databricks.
+
+Add the following configuration lines to `/etc/rstudio/rserver.conf` to use
+proxied authentication with Databricks and enable the administrator dashboard:
+
+```
+auth-proxy=1
+auth-proxy-user-header-rewrite=^(.*)$ $1
+auth-proxy-sign-in-url=<domain>/login.html
+admin-enabled=1
+```
+
+Add the following configuration line to `/etc/rstudio/rsession-profile` to set
+the `PATH` to be used with RStudio Server Pro:
+
+```
+export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:$PATH
+```
+
+Add the following configuration lines to `/etc/rstudio/rsession.conf` to
+configure sessions in RStudio Server Pro to work with Databricks:
+
+```
+session-rprofile-on-resume-default=1
+allow-terminal-websockets=0
+```
+
+Restart RStudio Server Pro:
+
+```
+sudo rstudio-server restart
+```
+
+## Access RStudio Server Pro
+
+From the Databricks console, click on your Databricks cluster:
+
+Click on the `Apps` tab:
+
+Click on the link to open RStudio Server Pro:
+
+If you configured proxied authentication in RStudio Server Pro as described in
```

Comment:
does the user need to previously exist for this to work?

Reply:
Good question!

Just tested this with another Databricks account, and it looks like the proxied auth and login work to create a home dir for other users accessing RSP in Databricks. So no need to provision users in RSP beforehand.

Added some text in b2a23e6 to clairfy this.